### PR TITLE
pylance-22-5-2024 -> pyright

### PR DESCRIPTION
**Pull request by bschnurr**
- Refactoring LS code
- Add callstacks on when parse results get cleaned.
-  Give encodeUri/decodeUri names that are easier to understand
- Remove unnecessary decode of CompletionItemData.uri
- Change the wording on number of files to analyze

### CODE REVIEW

**Review by rchiodo — APPROVED** [review]

**Review by debonte — APPROVED** [review]

File: packages/pyright-internal/typeshed-fallback/stubs/pyaudio/METADATA.toml [hunk 1]
```diff
@@ -4,6 +4,6 @@ version = "0.2.*"
 
 [tool.stubtest]
 # linux and win32 are equivalent
-platforms = ["darwin", "linux"]
+platforms = ["linux"]
 apt_dependencies = ["portaudio19-dev"]
-brew_dependencies = ["portaudio"]
```
Comment (bschnurr) [inline packages/pyright-internal/typeshed-fallback/stubs/pyaudio/METADATA.toml:9]: hmm. not sure why this is happening. pylance shouldn't be changing this

File: packages/pyright-internal/typeshed-fallback/stubs/uWSGI/METADATA.toml [hunk 2]
```diff
@@ -9,7 +9,4 @@ extra_description = """\
 """
 
 [tool.stubtest]
-# Run stubtest on MacOS as well, to check that the
-# uWSGI-specific parts of stubtest_third_party.py
-# also work there
-platforms = ["linux", "darwin"]
```
Comment (bschnurr) [inline packages/pyright-internal/typeshed-fallback/stubs/uWSGI/METADATA.toml:15]: hmm. not sure why this is happening. pylance shouldn't be changing this

Comment (rchiodo) [reply packages/pyright-internal/typeshed-fallback/stubs/uWSGI/METADATA.toml:15]: Yeah we should back these parts out. Or just manually do it.

Comment (bschnurr) [reply packages/pyright-internal/typeshed-fallback/stubs/uWSGI/METADATA.toml]: done

File: packages/pyright-internal/typeshed-fallback/stubs/pyaudio/METADATA.toml (hunk 1 above)
Comment (bschnurr) [reply packages/pyright-internal/typeshed-fallback/stubs/pyaudio/METADATA.toml]: removed

File: packages/pyright-internal/src/commands/createTypeStub.ts [hunk 3]
```diff
@@ -9,53 +9,64 @@
...
 import { AnalyzerServiceExecutor } from '../languageService/analyzerServiceExecutor';
 import { ServerCommand } from './commandController';
 import { Uri } from '../common/uri/uri';
+import { Workspace } from '../workspaceFactory';
 
 export class CreateTypeStubCommand implements ServerCommand {
-    constructor(private _ls: LanguageServerInterface) {}
+    constructor(private _ls: LanguageServerInterface) {
+        // Empty
+    }
```
Comment (bschnurr) [inline packages/pyright-internal/src/commands/createTypeStub.ts:21]: this also very strange

Comment (rchiodo) [reply packages/pyright-internal/src/commands/createTypeStub.ts:21]: Probably a hygiene issue?

Comment (bschnurr) [reply packages/pyright-internal/src/commands/createTypeStub.ts:21]: another bad merge

Comment (bschnurr) [reply packages/pyright-internal/src/commands/createTypeStub.ts:21]: actually it was a change in pylance, some formatting still seems strange though

**Pull request merged**